### PR TITLE
Added optional parameter --nsubgenes

### PR DESCRIPTION
**Pull request by jonoave**
Many thanks to contributing to qbic-pipelines!

Please fill in the appropriate checklist below (delete whatever is not relevant). These are the most common things requested on pull requests (PRs).

## PR checklist
 - [x] This comment contains a description of changes (with reason)
Added optional parameter `--nsubgenes`, due to potential errors with small dataset like miRNAs
https://www.biostars.org/p/456209/

 - [x] If you've fixed a bug or added code that should be tested, add tests!
 - [x] Ensure the test suite passes (`nextflow run . -profile test,docker`).
Tested with the following options, completed without errors:
```
nextflow run jonoave/rnadeseq -r add_feature_nsubgenes -profile test,cfc 
nextflow run jonoave/rnadeseq -r add_feature_nsubgenes -profile test,cfc --nsubgenes 500
```
 
- [ ] Documentation in `docs` is updated
Not updated.
 - [x] `CHANGELOG.md` is updated
Updated with line on v 1.4.1 
- [ ] `README.md` is updated
Not updated. 

**Learn more about contributing:** https://github.com/qbic-pipelines/rnadeseq/tree/master/.github/CONTRIBUTING.md

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # qbic-pipelines/rnadeseq: Changelog
 
+## 1.4.1 - dev
+
+### Added
+- Optional parameter "--nsubgenes" to change the number of subset genes for vst
+- Added a line mentioning the --nsubgenes value in the report output. 
+
```
Comment (d4straub) [inline CHANGELOG.md:8]: I propose adding this rather under the point 1.4.0 - dev below.

Comment (jonoave) [reply CHANGELOG.md]: Thanks Daniel, I just made the changes.

File: main.nf [hunk 2]
```diff
@@ -39,6 +39,7 @@ def helpMessage() {
       --relevel                     Tsv indicating list of factors (conditions in the metadata table) and the new level on which to relevel the factor. Check contrasts docs.
       --logFCthreshold              Threshold (int) to apply to Log 2 Fold Change to consider a gene as differentially expressed.
       --genelist                    Txt file with list of genes (one per line) of which to plot heatmaps for normalized counts across all samples.
+      --nsubgenes                   Number (int) of genes to subset in the vst step of deseq. Reduce default if encounter errors with smaller datasets
```
Comment (ggabernet) [inline main.nf:42]: Thanks for the addition and contribution, in order to make it clearer what exactly this parameter edits, I would suggest a clearer name such as `vst_genes_number`. What do you think?

Comment (ggabernet) [inline main.nf:42]:
```suggestion
      --vst_genes_number                   Number (int) of genes to subset for the vst step of DeSeq2. Reduce default if encountering errors with smaller datasets
```

**Review by ggabernet — COMMENTED** [review]
Hi @jonoave , thanks a lot for the helpful contribution. I just have a comment regarding the parameter name, which could be a bit more precise. Other than that it looks good!

File: main.nf (hunk 2 above)
Comment (jonoave) [reply main.nf]: Sure, sounds good. Thanks for reviewing this Gisela!

**Comment by d4straub** [comment]
I think the change that Gisela proposed needs to be also done in almost all other files (not in the R script though).

**Comment by jonoave** [comment]
> I think the change that Gisela proposed needs to be also done in almost all other files (not in the R script though).

Thanks for the reminder Daniel, I've made the changes to the file as well, and tested the rerun on cluster with 
`nextflow run jonoave/rnadeseq -r add_feature_nsubgenes_dev -profile test,cfc --vst_genes_number 500`

The pipeline completed successfully

File: assets/RNAseq_report.Rmd [hunk 3]
```diff
@@ -305,6 +307,7 @@ The raw count table and normalized count tables are available [here](./different
 The differential expression analysis is performed using the raw gene count table. 
 For PCA analysis and heatmap plotting, the regularized logarithm (rlog) normalized gene counts were employed.
 The vst (variant stabilizing transformation) normalized counts are also provided.
+The number of subset genes for the transformation is `r nsub_genes_text`.
```
Comment (ggabernet) [inline assets/RNAseq_report.Rmd:310]:
```suggestion
The number of subset genes for the `vst` transformation is `r nsub_genes_text`.
```

**Review by ggabernet — APPROVED** [review]
Hi @jonoave ,

thanks for the changes, looks good to me now, I just have a minor comment.

File: assets/RNAseq_report.Rmd (hunk 3 above)
Comment (jonoave) [reply assets/RNAseq_report.Rmd]: Thanks Gisela, I've made the text change to RNAseq_report.Rmd

File: docs/usage.md [hunk 4]
```diff
@@ -5,44 +5,58 @@
...
-    - [`default`](#default)
+  - [Contrasts](#contrasts)
+    - [Default](#default)
     - [`--relevel`](#--relevel)
     - [`--contrast_matrix`](#--contrast_matrix)
     - [`--contrast_list`](#--contrast_list)
-    - [`--contrast_pairs`](#--contrast_pairs)
-- [Optional arguments](#Optional-arguments)
-    - [`--logFCthreshold`](#--logFCthreshold)
+    - [`--contrast_pairs``](#--contrast_pairs)
```
Comment (ggabernet) [inline docs/usage.md:27]:
```suggestion
    - [`--contrast_pairs`](#--contrast_pairs)
```

**Pull request merged**